### PR TITLE
Update CONTRIBUTING.md, update README, and remove docs folder

### PR DESCRIPTION
**Pull request by mxbclang**
## Summary

Makes several changes to documentation due to the addition of the new [Handbook on the Core Performance site](https://make.wordpress.org/performance/handbook/), including:

- Updating `README` to remove Quick Start, Useful commands, and Docs sections and instead link to relevant URLs in Handbook
- Updating `CONTRIBUTING.md` to remove old content, add link to relevant URL in Handbook, and add details about WP and PHP minimum requirements
- Removing the `/docs` folder as this content has all been moved to the Handbook

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,10 +1,14 @@
-# Contributing Guidelines
+# Contributing to the Performance Lab plugin
 
-Welcome to the Performance plugin project! We're glad you're here. Here's a few things to help you get started.
+Thank you for your interest in contributing to the Performance Lab plugin! Complete documentation for getting started and contributing to the plugin is available in the [Performance Lab](https://make.wordpress.org/performance/handbook/welcome/performance-lab/) section of our [Performance Team Handbook](https://make.wordpress.org/performance/handbook/welcome/performance-lab/).
```
Comment (felixarntz) [inline CONTRIBUTING.md]: The second link here should probably point to the overall handbook (`https://make.wordpress.org/performance/handbook/`) instead.

**Review by felixarntz — CHANGES_REQUESTED** [review]
@bethanylang LGTM, except for 2 things:
* There are changes to `package-lock.json` in this PR that somehow made it in, but shouldn't be here.
* See the one individual comment below for my only content suggestion.

**Comment by mxbclang** [comment]
@felixarntz Handbook link fixed and I _think_ I successfully removed `package-lock.json` from the PR, but I'm also not the greatest with GitHub, so please let me know if I broke something!

**Comment by felixarntz** [comment]
@bethanylang I fixed it now :)

**Review by felixarntz — APPROVED** [review]
@bethanylang LGTM!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,10 +1,14 @@
-# Contributing Guidelines
+# Contributing to the Performance Lab plugin
 
-Welcome to the Performance plugin project! We're glad you're here. Here's a few things to help you get started.
+Thank you for your interest in contributing to the Performance Lab plugin! Complete documentation for getting started and contributing to the plugin is available in the [Performance Lab](https://make.wordpress.org/performance/handbook//performance-lab/) section of our [Performance Team Handbook](https://make.wordpress.org/performance/handbook/).
```
Comment (mukeshpanchal27) [inline CONTRIBUTING.md]:
```suggestion
Thank you for your interest in contributing to the Performance Lab plugin! Complete documentation for getting started and contributing to the plugin is available in the [Performance Lab](https://make.wordpress.org/performance/handbook/performance-lab/) section of our [Performance Team Handbook](https://make.wordpress.org/performance/handbook/).
```

**Review by mukeshpanchal27 — APPROVED** [review]
Thanks, @bethanylang. One nit-picking issue

Can you please check the handbook document URL? I discovered one section with a doc folder URL at https://make.wordpress.org/performance/handbook/get-involved/#attend-a-release-party.

File: CONTRIBUTING.md (hunk 2 above)
Comment (mxbclang) [reply CONTRIBUTING.md]: @mukeshpanchal27 Good catch! Should be fixed now.

**Pull request merged**